google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITForeignKeyDeleteCascadeTest.java

```diff
@@ -0,0 +1,582 @@
+          + ") PRIMARY KEY(VenueId, SingerId)";
+
+  private static final String POSTGRES_CREATE_TABLE_CONCERT_V2_WITHOUT_FOREIGN_KEY =
+      "CREATE TABLE ConcertsV2 (\n"
+          + "      venue_id      BIGINT NOT NULL,\n"
+          + "      singer_id     BIGINT NOT NULL,\n"
+          + "      PRIMARY KEY (venue_id, singer_id)\n"
+          + "      )";
+
+  private static final String ALTER_TABLE_CONCERT_V2_WITH_FOREIGN_KEY =
+      "ALTER TABLE ConcertsV2 "
+          + "ADD CONSTRAINT Fk_Concerts_Singer_V2 FOREIGN KEY(SingerId) REFERENCES Singer(SingerId) "
+          + "ON DELETE CASCADE";
+
+  private static final String POSTGRES_ALTER_TABLE_CONCERT_V2_WITH_FOREIGN_KEY =
+      "ALTER TABLE ConcertsV2 "
+          + "ADD CONSTRAINT Fk_Concerts_Singer_V2 FOREIGN KEY(singer_id) REFERENCES Singer(singer_id) "
+          + "ON DELETE CASCADE";
+
+  private static final String ALTER_TABLE_CONCERT_V2_UPDATE_FOREIGN_KEY_WITHOUT_DELETE_CASCADE =
+      "ALTER TABLE ConcertsV2 "
+          + "ADD CONSTRAINT Fk_Concerts_Singer_V2 FOREIGN KEY(SingerId) REFERENCES Singer(SingerId) ";
+
+  private static final String
+      POSTGRES_ALTER_TABLE_CONCERT_V2_UPDATE_FOREIGN_KEY_WITHOUT_DELETE_CASCADE =
+          "ALTER TABLE ConcertsV2 "
+              + "ADD CONSTRAINT Fk_Concerts_Singer_V2 FOREIGN KEY(singer_id) REFERENCES Singer(singer_id) ";
+
+  private static final String ALTER_TABLE_CONCERT_V2_DROP_FOREIGN_KEY_CONSTRAINT =
+      "ALTER TABLE ConcertsV2\n" + "DROP CONSTRAINT Fk_Concerts_Singer_V2";
+
+  private static Database GOOGLE_STANDARD_SQL_DATABASE;
+  private static Database POSTGRESQL_DATABASE;
+  private static List<Database> dbs = new ArrayList<>();
+
+  @Parameterized.Parameter(0)
+  public DialectTestParameter dialect;
+
+  @BeforeClass
+  public static void setUpDatabase() {
+    if (!EmulatorSpannerHelper.isUsingEmulator()) {
+      GOOGLE_STANDARD_SQL_DATABASE =
+          env.getTestHelper()
+              .createTestDatabase(
+                  ImmutableList.of(CREATE_TABLE_SINGER, CREATE_TABLE_CONCERT_WITH_FOREIGN_KEY));
+      POSTGRESQL_DATABASE =
+          env.getTestHelper()
+              .createTestDatabase(
+                  Dialect.POSTGRESQL,
+                  ImmutableList.of(
+                      POSTGRES_CREATE_TABLE_SINGER,
+                      POSTGRES_CREATE_TABLE_CONCERT_WITH_FOREIGN_KEY));
+
+      dbs.add(GOOGLE_STANDARD_SQL_DATABASE);
+      dbs.add(POSTGRESQL_DATABASE);
+    }
+  }
+
+  @AfterClass
+  public static void tearDown() {
+    for (Database db : dbs) {
+      db.drop();
+    }
+    dbs.clear();
+  }
+
+  @Test
+  public void testForeignKeyDeleteCascadeConstraints_withCreateDDLStatements() {
+    assumeFalse(
+        "Emulator does not yet support foreign key delete cascade",
+        EmulatorSpannerHelper.isUsingEmulator());
+
+    final DatabaseClient databaseClient = getCreatedDatabaseClient();
+    final String referentialConstraintQuery = getReferentialConstraintsQueryStatement();
```

Comment:
Here and elsewhere:

I think you should:
1. Remove the `getReferentialConstraintsQueryStatement()` method and directly have the SQL statement here. That makes it easier to read what this test case does. AFAICT, it is only used twice, so the code duplication is minimal.
2. The query itself does not need to be dialect specific. Both dialects support using single quotes for string literals. (That is; both dialects can use `select col1, col2, ... from my_table where col1='some value'`).
3. It might be even better to change the query into using a query parameter. That would require the query to be slightly different per dialect, but only for the name of the query parameter (that is; `$1` vs `@p1`). Generating a query parameter could be put in a separate method, so you don't need that part in the test method itself.